ledger/complete/mtrie/trie/trie.go

```diff
@@ -352,6 +352,15 @@ func NewTrieWithUpdatedRegisters(
 	if err != nil {
 		return nil, 0, fmt.Errorf("constructing updated trie failed: %w", err)
 	}
+
+	if prune {
```

Comment:
This check doesn't seem right.   The IsDefaultNode check should be part of NewMTrie (we should unconditionally set MTrie's root to nil if the updatedRoot is a default node)

(aside: I would probably remove the prune argument from `update` altogether.  It doesn't save us much computation wise and the resulting tree structure is not fully deterministic)